Packs/OracleCloudInfrastructure/Integrations/OracleCloudInfrastructureEventCollector/OracleCloudInfrastructureEventCollector.py

```diff
@@ -0,0 +1,387 @@
+                fingerprint=key_fingerprint,
+                private_key_content=validated_private_key,
+                private_key_file_location=None,
+            )
+        except Exception as e:
+            raise DemistoException(
+                'Could not create a valid OCI singer object, Please check the instance configuration parameters.',
+                exception=e,) from e
+
+        return singer
+
+    def build_audit_base_url(self, region: str) -> str:
+        """Build the base URL for the client.
+
+        Args:
+            region (str): Region parameter.
+
+        Raises:
+            DemistoException: If the region is not valid.
+
+        Returns:
+            str: Base URL for the client.
+        """
+        if not is_region(region):
+            raise DemistoException('Could not create a valid OCI configuration dictionary fou to invalid region parameter. \
+                Please check your OCI related instance configuration parameters.')
+
+        return f'https://audit.{region}.oraclecloud.com/{PORT}/auditEvents'
+
+    def validate_private_key_syntax(self, private_key_parameter: str) -> str:
+        """Validate private key parameter syntax.
+        The Private Key parameter needs to be provided to the OCI SDK singer object in a specific format.
+        The most common way to obtain the private key is to download a .pem file from the OCI console.
+        If copied from a .pem file, the private key parameter may contain unnecessary spaces.
+        Further more, since the format uses \n as part of the key,
+        passing this value as a configuration parameter may result in escaped \n characters.
+
+        This function will preform the following actions on the private key parameter:
+            - Unescape the string.
+            - Remove unnecessary spaces in the string.
+
+        Example:
+        Raw Private Key parameter: -----BEGIN PRIVATE KEY-----\\n\n THIS-IS-A\\n\n PRIVATE-KEY\\n\n -----END PRIVATE KEY-----
+        Output: -----BEGIN PRIVATE KEY-----\nTHIS-IS-A\nPRIVATE-KEY\n-----END PRIVATE KEY-----
+
+        Args:
+            private_key (str): Private Key parameter.
+
+        Returns:
+            str: Private Key parameter unescaped and spaceless.
+        """
+        private_key = stringUnEscape(private_key_parameter)
+        private_key = private_key.replace('\n\n', '\n')
+
+        if ' ' not in private_key:
+            return private_key
+
+        prefix = '-----BEGIN PRIVATE KEY-----\n'
+        postfix = '\n-----END PRIVATE KEY-----'
+
+        private_key = private_key.replace(prefix, '').replace(postfix, '')
+
+        private_key_sections = private_key.strip().split(' ')
+        striped_private_key = ''.join(private_key_sections)
+        return prefix + striped_private_key + postfix
+
+
+''' Event related functions '''
+
+
+def add_time_key_to_events(events: List[dict[str, Any]]) -> List[dict[str, Any]]:
+    """
+    Add the _time key to the events.
```

Comment:
Remind me why it is needed? 
Might be worth explaining that here.

Reply:
It is the current decided solution for the `_time` data model field.
It used to be parsed as a parsing rule in XSIAM, but to avoid a lot of cases where parsing rules were created only for this `_time` field, now the solution is to "map" this attribute programatically.

Reply:
Is it a general solution for all data collectors? or is it something specific for this integration?

Reply:
Its the current acceptable solution for event collectors.